### PR TITLE
fix: better error for missing web crypto

### PR DESCRIPTION
**Pull request by alanshaw**
This PR simply detects missing web crypto and throws an error with an appropriate message.

This is a stepping stone that will help users understand the problem until we have time to do a refactor of this module and of all the modules that use it to enable optionally passing your own crypto implementation.

refs https://github.com/libp2p/js-libp2p-crypto/pull/149
refs https://github.com/libp2p/js-libp2p-crypto/pull/150
refs https://github.com/libp2p/js-libp2p-crypto/issues/105
refs https://github.com/ipfs/js-ipfs/issues/2153
refs https://github.com/ipfs/js-ipfs/issues/2017

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,11 +51,32 @@ This repo contains the JavaScript implementation of the crypto primitives needed
...
+// etc.
+//
+// See full API details below...
+```
+
+### Web Crypto API
+
+The `libp2p-crypto` library depends on the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) in the browser. Web Crypto is available in all modern browsers, however browsers restrict its usage to [Secure Contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts).
+
+**This means you will not be able to use some `libp2p-crypto` functions in the browser when the page is served over HTTP.**
```
Comment (dirkmc) [inline README.md]: I suggest adding a couple of words to make it clear that you *can* serve over HTTPS

**Comment by jacobheun** [comment]
I rebased this with the latest master so we can include this and the error code update in the next release.

**Comment by alanshaw** [comment]
@dignifiedquire 🙏 can we get a merge and release?

**Comment by alanshaw** [comment]
❤️ thanks @dignifiedquire

**Pull request merged**